src/cljc/matcher_combinators/core.cljc

```diff
@@ -460,15 +476,43 @@
       :else
       {::result/type  :mismatch
        ::result/value (model/->Mismatch desc actual)
-       ::result/weight 1})))
+       ::result/weight 1}))
+  (-name [_] 'predicate))
+
+(defn- printable-matcher [matcher]
+  (try
+    (if-let [n (-name matcher)]
+      `(~(symbol n) ~(:expected matcher))
+      matcher)
+    (catch IllegalArgumentException _e
+      matcher)))
+
+(defrecord Mismatcher
+  [expected]
+  Matcher
+  (-matcher-for [this] this)
+  (-matcher-for [this _] this)
+  (-match [this actual]
+    (let [{::result/keys [type weight] :as result} (match expected actual)]
+      (if (= :match type)
+        {::result/type   :mismatch
+         ::result/value  (model/->ExpectedMismatch
+                          (printable-matcher expected)
+                          actual)
```

Comment:
this is needed to be able to print a mismatch message that includes the original matcher when a mismatch is expected. Like the `(embeds [odd?])` in the example below
```
FAIL in (absent-matcher) (matchers_test.clj:386)
assert an entry is definitely not in a sequence
expected: (match? (m/mismatch (m/embeds [odd?])) [1 3 5 7])
  actual: (mismatch
 expected mismatch from:
 (embeds [#function[clojure.core/odd?]])
 [1 3 5 7])
```